src/scripts/achievements/AchievementHandler.ts

```diff
@@ -33,12 +33,19 @@ class AchievementHandler {
         return this.achievementListFiltered().slice(this.navigateIndex() * 10, (this.navigateIndex() * 10) + 10);
     }
 
-    public static filterAchievementList() {
+    public static filterAchievementList(retainPage?: boolean) {
         this.achievementListFiltered(this.achievementList.filter(a => a.region <= player.highestRegion() &&
                                     (this.filter.status() == 'all' ? true : a.unlocked == JSON.parse(this.filter.status())) &&
                                     (this.filter.type()   == 'all' ? true : a.property.constructor.name == this.filter.type()) &&
                                     (this.filter.region() == 'all' ? true : a.region == +this.filter.region())));
-        this.resetPages();
+        if (!retainPage) {
+            this.resetPages();
+        } else if (this.getAchievementListWithIndex().length == 0) {
+            if (this.navigateIndex() > 0) {
+                this.calculateNumberOfTabs();
+                this.navigateIndex(this.navigateIndex() - 1);
+            }
```

Comment:
This is great! We can make it just a bit cleaner by combining the if statement and reusing the `numberOfTabs()` value that you updated
```suggestion
        } else if (this.getAchievementListWithIndex().length == 0 && this.navigateIndex() > 0) {
            this.calculateNumberOfTabs();
            this.navigateIndex(this.numberOfTabs())
```